### DISCUSSION TITLE
connect switch 之后form.values中没有对应的字段值

### DISCUSSION THREAD

**Discussion by happyElina** [alibaba/formily#1904] [Q&A 2.x] [answered]
在准备将公司的Toggle(也就是switch)组件connect到formily中之后，发现一开始form.values中没有对应的字段值，在点击切换toggle之后就出现了对应的值，封装的过程中也将value 和 checked 映射了
`export const FormToggle = connect(
  Toggle,
  mapProps(
    {
      value: 'checked',
    },
    (props, field) => {
      delete props['value'];
      return {
        ...props,
      };
    }
  ),
  mapReadPretty(PreviewText.Toggle)
);

export default FormToggle;`

替换成@formily/antd 中的switch也是这样的,demo为：
`import React from 'react'
import { createForm } from '@formily/core'
import { createSchemaField, FormConsumer } from '@formily/react'
import { Form, FormItem,Switch } from '@formily/antd'
import 'antd/dist/antd.css'
const form = createForm()

const SchemaField = createSchemaField({
  components: {
    FormItem,
    Switch,
  },
})

const schema = {
  type: 'object',
  properties: {
    toggle: {
      title: 'toggle',
      type: 'string',
      'x-decorator': 'FormItem',
      'x-component': 'Switch',
      'x-component-props': {
        checked: true,
      }
    },
  },
}

const App =  () => (
  <div style={{padding: '20px'}}>
    <Form form={form} labelCol={6} wrapperCol={10}>
      <SchemaField schema={schema} />
      <code>
        <pre>
          <FormConsumer>
            {(form) => JSON.stringify(form.values, null, 2)}
          </FormConsumer>
        </pre>
    </code>
    </Form>
  </div>

)

export default App;`

后来发现在createForm中增加values默认值或者initialValues 就可以了，请问这种问题就必须是给一个values默认值或者initialValues吗？如果没有给到默认值，如果表单中的toggle没有被操作修改过，提交的时候还是会缺少form.values中对应的值？

**Comment by atzcl** [comment, marked as answer]
这种场景一般都会让设置默认值的吧

```ts
const schema = {
  type: 'object',
  properties: {
    toggle: {
      title: 'toggle',
      type: 'boolean',
      default: true, // or false
      'x-decorator': 'FormItem',
      'x-component': 'Switch',
    },
  },
}
```

**Reply by happyElina** [reply]
get
